### ISSUE TITLE
.NET Core 1.0.1 with SQLitePCLRaw.bundle_e_sqlite3 fails to initialize

### ISSUE THREAD

**Issue by samcragg** [ericsink/SQLitePCL.raw#100]
First of all, great library that exposes the low level stuff - it's great to be able to look at the SQLite documentation and be able to apply that in C#.

I'm having a problem creating a .NET Core console app, which I think should be supported? I'm running the following under Visual Studio 2015 (14.0.25431.01 Update 3) on Windows 10 (64 bit):

    using SQLitePCL;
    
    namespace SqliteCore
    {
        public class Program
        {
            public static void Main(string[] args)
            {
                Batteries.Init();
                sqlite3 db;
                raw.sqlite3_open("database.db", out db);
            }
        }
    }

The above works fine in a console app targeting .NET Framework 4.6, however, in a .NET Core 1.0.1 console app I get an exception `You need to call SQLitePCL.raw.SetProvider().  If you are using a bundle package, this is done by calling SQLitePCL.Batteries.Init().`. Do I need to put anything else in the `project.json` file for it to be able to pick up the right DLL at runtime? Here's my current one, which is just the VS template plus the reference to the e_sqlite bundle:

    {
      "version": "1.0.0-*",
      "buildOptions": {
        "emitEntryPoint": true
      },
    
      "dependencies": {
        "Microsoft.NETCore.App": {
          "type": "platform",
          "version": "1.0.1"
        },
        "SQLitePCLRaw.bundle_e_sqlite3": "1.0.0"
      },
    
      "frameworks": {
        "netcoreapp1.0": {
          "imports": "dnxcore50"
        }
      }
    }

Thanks for you help

[SqliteCore.zip](https://github.com/ericsink/SQLitePCL.raw/files/479296/SqliteCore.zip)

**Comment by ericsink** [comment]
Hmmm.  Yes, I think this should work.

But I'm not surprised it doesn't, since I didn't test with .NET Core.  I will do that and see what kind of fixes I need to make.

My first guess is that I bet the bundle isn't smart enough (yet?) to handle .NET Core.  

If you want to try something on your own, I wonder if it would work by doing the bundle stuff manually?

Add these packages:

SQLitePCLRaw.core
SQLitePCLRaw.provider.e_sqlite3.netstandard11
SQLitePCLRaw.lib.e_sqlite3.v140

And instead of calling Batteries.Init(), call:

raw.SetProvider(new SQLite3Provider_e_sqlite3())

This has a somewhat better chance of working right now.

I'll post another update after I look at this more closely.

Thanks for the report.

**Comment by samcragg** [comment]
Think we're getting closer, now I get an exception on the SetProvider line of `DllNotFoundException: Unable to load DLL 'e_sqlite3': The specified module could not be found. (Exception from HRESULT: 0x8007007E)`

Looks like it might be an issue where the DLL isn't being copied to the output directory (must admit there's still lots of voodoo with how that works in .NET Core)?

**Comment by ericsink** [comment]
Yeah, the download of .NET Core on my home internet line went faster than I figured, so I just went ahead and tried the same thing.  And got the same error.

So the necessary fixes here will go beyond just the bundle.  It looks like maybe the targets-file approach used in SQLitePCLraw.lib.* is not compatible with xproj/project.json.  I'll probably need to go ahead and move the native files into runtimes/(rid)/native in the nuget package, which I planned to do anyway.

Even after I fix the lib package, the bundle will be interesting.  I want this to work, but nuget v3 runtime-specific dependencies are not supported yet, so it's possible the bundle's behavior for .NET Core might not be as elegant as I would like.  We'll see.
